htdocs/compta/prelevement/class/bonprelevement.class.php

```diff
@@ -1209,7 +1264,7 @@ public function create($banque = 0, $agence = 0, $mode = 'real', $format = 'ALL'
 
 				dol_syslog(__METHOD__." Init direct debit or credit transfer file for ".count($factures_prev)." invoices", LOG_DEBUG);
 
-				if (count($factures_prev) > 0) {
+				if (count($factures_prev) > 0 || !empty($sourcetype)) {
```

Comment:
Do we really need the test 
|| !empty($sourcetype)
here ?
If we have $factures_prev with document to pay, this may be enough. And if there is nothing, it means we have no salary to pay so no need to enter. Am i right ?